tls/s2n_prf.h

```diff
@@ -26,21 +26,23 @@
 /* Enough to support TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, 2*SHA384_DIGEST_LEN + 2*AES256_KEY_SIZE */
 #define S2N_MAX_KEY_BLOCK_LEN 160
 
-struct p_hash_state {
+/* TODO: make this a union. If s2n is in FIPS mode it's always going to use evp_hmac, and outside of FIPS it should
+ * always use s2n_hmac */
```

Comment:
What's blocking this TODO? Is there an issue for it?

Reply:
Nothing, it would just make this PR even bigger. I created https://github.com/aws/s2n-tls/issues/3111 for follow up.